File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    map_1 = []
    map_2 = []
    result=[]
    b=''
    for i in arr1:
        map_1.append(bin(i))
    for j in arr2:
        map_2.append(bin(j))
        
    for k in range(len(map_1)):
        answer.append(int(map_1[k],2) | int(map_2[k],2))
        
    for i in range(len(answer)):
        answer[i] = bin(answer[i])
    for i in answer:
        if len(i)-2<n:
            b+=' '*(n-(len(i)-2))
        for j in range(2,len(i)):
            if i[j] == '1':
                b+='#'
            elif i[j] == '0':
                b+=' '
        result.append(b)
        b=''
    return result

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        String[] array1 = new String[n];
        String[] array2 = new String[n];
        
        for (int i=0; i<n; i++){
            array1[i] = num_to_bit(arr1[i], n);
            array2[i] = num_to_bit(arr2[i], n);
        }
        
        for (int i = 0; i < n; i++) {
            StringBuilder sb = new StringBuilder();
            for (int j = 0; j < n; j++) {
                if (array1[i].charAt(j) == '#' || array2[i].charAt(j) == '#') {
                    sb.append("#");
                } else {
                    sb.append(" ");
                }
            }
            answer[i] = sb.toString();
        }
        return answer;
    }
    
    public static String num_to_bit(int number, int range){
        String returnBit = "";
        while(number > 0){
            if (number % 2 == 0){
                returnBit = " " + returnBit;
            }
            else{
                returnBit = "#" + returnBit;
            }
            number /= 2;
        }
        while (returnBit.length() < range){
            returnBit = " " + returnBit;
        }
        return returnBit;
    }
}